### ISSUE TITLE
Possible bug: Hover tool does not work with filtered source

### ISSUE THREAD

**Issue by GenericUser123** [bokeh/bokeh#6955]
bokeh 0.12.8rc2, Tornado 4.5.1, Windows 10, Firefox 55.0.3, python 3.6.1

A simple ColumnDataSource with 5 example points, splittet up into 3 "groups" ("A", "B" and "C") is filtered using a select widget.
The hover tool of the generated plot works for the points of group "A", which is the default value of the select widget. Selecting group "B" or "C", the points are plotted correctly, but the hover tool will not work anymore.

I tried to change the default value to "B", but the hover tool will still only works with group "A" and not with "B" or "C", so it does not seem to depend on the default value of the select widget.

I guess that might be a bug?
Is there any workaround?

Thanks a lot and best regards,
Juerg

```
# code goes here between backticks
from bokeh.plotting import figure
from bokeh.models import ColumnDataSource, CDSView, GroupFilter
from bokeh.models.widgets import Select
from bokeh.layouts import row, widgetbox
from bokeh.io import curdoc


data = {'x_values': [1, 2, 3, 4, 5],
        'y_values': [1, 1, 0, 0, 1],
        'cat': ["A", "A", "B", "B", "C"]}


cat_filter = Select(title='category', options=["A", "B", "C"], value="A")
source = ColumnDataSource(data=data)


def create_plot():
    view1 = CDSView(source=source, filters=[GroupFilter(column_name='cat', group=cat_filter.value)])
    p=figure(tools="hover")
    p.circle("x_values","y_values", size=20, source=source, view=view1)
    return p


def update(attr, old, new):
    l.children[1]=create_plot()
    
cat_filter.on_change('value', update)


inputs=widgetbox(cat_filter)
l=row(inputs,create_plot())
update

curdoc().add_root(l)

```

**Comment by aavanian** [comment]
I get the same (although using CustomJS callbacks). Color mapping also seems buggy:

Snippet:
```python
import random

from bokeh.layouts import widgetbox
from bokeh.models.mappers import LinearColorMapper
from bokeh.palettes import Viridis256
from bokeh.layouts import row, column
from bokeh.plotting import figure, show, output_notebook
from bokeh.models import ColumnDataSource, CDSView, GroupFilter, CustomJS, Select, HoverTool

data = {
    'age': [i for i in range(100)],
    'label': [random.choice(['a', 'b', 'c']) for _ in range(100)],
    'y': [random.uniform(0, 1) for _ in range(100)]
}
cds = ColumnDataSource(data)

start = 'a'

cm = LinearColorMapper(Viridis256, high=100, low=0)
transform = {'field': 'age', 'transform': cm}

filter_ = GroupFilter(column_name='label', group=start)
view = CDSView(source=cds, filters=[filter_])

p_unfiltered = figure(toolbar_location='above')
scatter = p_unfiltered.circle(x='age', y='y', source=cds, fill_color=transform, line_color=None, size=10, alpha=0.5)
ht = HoverTool(
    renderers=[scatter],
    tooltips=[('age', '@age'), ('y', '@y'), ('label', '@label')],
    formatters={'date': 'datetime'},
    mode='mouse'
)
p_unfiltered.add_tools(ht)

p_filtered = figure(x_range=p_unfiltered.x_range, y_range=p_unfiltered.y_range, toolbar_location='above')
scatter = p_filtered.circle(x='age', y='y', source=cds, view=view, fill_color=transform, line_color=None, size=10, alpha=0.5)
ht = HoverTool(
    renderers=[scatter],
    tooltips=[('age', '@age'), ('y', '@y'), ('label', '@label')],
    formatters={'date': 'datetime'},
    mode='mouse'
)
p_filtered.add_tools(ht)

cluster_callback = CustomJS(args=dict(fil=filter_, source=cds), code="""
            var cluster = cb_obj.value;
            fil.group = cluster;
            source.change.emit();
""")

cluster_select = Select(title="Cluster", value=start,
                      options=['a', 'b', 'c'],
                      callback=cluster_callback
                      )

controls = widgetbox([cluster_select])

layout = row([p_unfiltered, controls, p_filtered])
show(layout)
```
![image](https://user-images.githubusercontent.com/13487305/30908664-df3d22c4-a3b0-11e7-81aa-4c5c5dcaf608.png)

**Comment by bryevdv** [comment]
cc @clairetang6

**Comment by clairetang6** [comment]
This is fixed by [PR #6967](https://github.com/bokeh/bokeh/pull/6967) and the color mapping issue by [PR #6966](https://github.com/bokeh/bokeh/pull/6966)!
